.github/workflows/build.yaml

```diff
@@ -3,12 +3,25 @@ name: Build
 on: [push, pull_request]
 
 jobs:
-  build:
+  test:
     runs-on: ubuntu-latest
-    env:
-      # disable the demon, as it's causing some problems
-      GRADLE_OPTS: -Dorg.gradle.daemon=false
+    strategy:
+      matrix:
+        version: [ 8, 9, 11, 17 ]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK 1.8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
 
+      - name: Build and test the project
+        env:
+          JAVA_VERSION: ${{ matrix.version }}
+        run: ./gradlew check javadoc
```

Comment:
Any reason we're introducing a new job here, instead of changing the existing one?

Reply:
None at all, will collapse them.